lib/request-shim.js

```diff
@@ -52,7 +52,7 @@ function getOptions(options) {
         searchParams: options.searchParams || options.qs,
         headers: options.headers,
         throwHttpErrors: options.throwHttpErrors || false,
-        retry: options.retry || 0,
+        retry: options.retry || 5,
```

Comment:
Taken into account this `retry` parameter, I wonder if the loop based in `retry_count` is really needed. Could you include got library documentation about this `retry` option, please?

Reply:
Hi @fgalan , I include this retry count to make the default value less than httpOptions.retry so that the agent can retry 5 times before error.

>Same here in https://github.com/telefonicaid/iotagent-json/blob/ef782f8f89ea2d0769dc36ecf9d96c63ef82cc17/lib/bindings/AMQPBinding.js#L130
and numRetried is used.

Reply:
I mean, modifications from L98 to L130 wouldn't be necessary, as far as I understand.

Reply:
Hi @fgalan , I have changed the default value but the agent was not retrying to connect to orion CB when it was down. Thats why I have to code to make it retry 5 times and also through an error as it was initially doing but after retrying 5 times after a particular interval of time. Thanks for the reply.

Reply:
Do you mean that got library is not working properly? As far as I understand, the got library should deal with this.

https://github.com/sindresorhus/got/blob/main/documentation/7-retry.md shows a pretty sophisticated configuration. I'd suggest to see how it works and use it.

Reply:
Hi @fgalan , I just want to say to make that work we need some value to make it work 5 times so that it comes out of loop after max retries as it is implemented also in 
>https://github.com/telefonicaid/iotagent-json/blob/ef782f8f89ea2d0769dc36ecf9d96c63ef82cc17/lib/bindings/AMQPBinding.js#L130

and line ```145```